packages/ruby/lib/readme/metrics.rb

```diff
@@ -34,7 +37,15 @@ def call(env)
       response = Rack::Response.new(body, status, headers)
 
       har = Har::Serializer.new(env, response, start_time, end_time, @filter)
-      user_info = @get_user_info.call(env)
+
+      begin
+        user_info = @get_user_info.call(env)
+        raise if user_info_invalid?(user_info)
+      rescue
+        puts Errors::BadBlockMessage.call(user_info)
+        return [status, headers, body]
+      end
```

Comment:
I'm not sure we need a `begin/end` block here if we're manually raising? This feels like using exceptions for control-flow. Thoughts on expressing this as a conditional instead?

```ruby
user_info = @get_user_info.call(env)

if user_info_invalid?(user_info)
  puts Errors::BadBlockMessage.call(user_info)
else
  payload = Payload.new(har, user_info, development: @development)
  @@request_queue.push(payload.to_json)
end

[status, headers, body]
```

I could see extracting the conditional into a private method so `call` doesn't get too cluttered

Reply:
Part of what I wanted to do here was catch the situation where the block itself raises an exception so the application can continue running. Does that make sense to do?

Reply:
I see what you're saying. I kind of feels like feature creep though? We have a [separate card ](https://trello.com/c/MFKwi9k7/14-if-a-logging-request-fails-it-should-not-impact-the-client-api-request)for not crashing the client request if the middleware has an issue. I think we'll want a single rescue around the whole process. 

If the block raises, then we can't send a valid payload to Readme so I don't think there is much value to adding a local rescue just for the block.

Reply:
Great that makes sense. I'll update this to not use rescue block.